doc/articles/Asynchronous Operations.md

```diff
@@ -0,0 +1,68 @@
+
+> @copydoc ::WGPUCallbackMode_WaitAnyOnly
+> @copydoc ::WGPUCallbackMode_AllowProcessEvents
+> @copydoc ::WGPUCallbackMode_AllowSpontaneous
+
+## wgpuInstanceWaitAny {#Wait-Any}
+`WGPUWaitStatus wgpuInstanceWaitAny(WGPUInstance, size_t futureCount, WGPUFutureWaitInfo * futures, uint64_t timeoutNS)`
+
+Waits on any WGPUFuture in the list of `futures` to complete for `timeoutNS` nanoseconds. Returns when at least one `WGPUFuture` is completed or `timeoutNS` elapses, whichever is first. If `timeoutNS` is zero, all `futures` are polled once, without blocking.
+
+Returns `::WGPUWaitStatus_Success` if at least one `WGPUFuture` completes. WGPUFutureWaitInfo::completed is set to true for all completed futures. See @ref WGPUWaitStatus for other status codes.
+
+Within this call, for any `WGPUFuture`s that completed, their respective callbacks will fire.
+
+### Timed Wait {#Timed-Wait}
+
+Use of _timed waits_ (`timeoutNS > 0`), must be enabled on the WGPUInstance with WGPUInstanceFeatures::timedWaitAnyEnable, and the number of futures waited on must be less than or equal to WGPUInstanceFeatures::timedWaitAnyMaxCount. Supported instance features may be queried using `::wgpuGetInstanceFeatures`.
+
+### Mixed Sources {#Mixed-Sources}
+
+Asynchronous operations may originate from different sources. There are CPU-timeline operations and there are Queue-timeline operations. Within a _timed wait_, it is only valid to wait on `WGPUFuture`s originating from a single `WGPUQueue`. Waiting on two futures from different queues, or waiting on a Queue-timeline future and some other CPU-timeline future is an error.
+
+#### CPU-Timeline Operations
+ - ::wgpuInstanceRequestAdapter
+ - ::wgpuAdapterRequestDevice
+ - ::wgpuShaderModuleGetCompilationInfo
+ - ::wgpuDeviceCreateRenderPipelineAsync
+ - ::wgpuDeviceCreateComputePipelineAsync
+ - ::wgpuDevicePopErrorScope
+
+#### Queue-Timeline Operations
+ - ::wgpuBufferMapAsync
+ - ::wgpuQueueOnSubmittedWorkDone
+
+## wgpuInstanceProcessEvents {#Process-Events}
+`void wgpuInstanceProcessEvents(WGPUInstance)`
+
+Processes asynchronous events on this `WGPUInstance`, calling any callbacks for asynchronous operations created with `::WGPUCallbackMode_AllowProcessEvents` that have completed. This is a non-blocking operation.
+
+## Device Events
+
+Device events are slightly different in that their callback info (`WGPUDeviceLostCallbackInfo` and `WGPUUncapturedErrorCallbackInfo`) are passed on the `WGPUDeviceDescriptor`, instead of in a function argument. There is no `WGPUFuture` returned for either callback.
```

Comment:
IIRC we said there would be a getter for the DeviceLost future, add TODO about that?

Reply:
Added TODO